guides/micronaut-mongodb-synchronous/java/src/main/java/example/micronaut/MongoDbFruitRepository.java

```diff
@@ -0,0 +1,69 @@
+package example.micronaut;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import io.micronaut.core.annotation.NonNull;
+import jakarta.inject.Singleton;
+import org.bson.Document;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Singleton // <1>
+public class MongoDbFruitRepository implements FruitRepository {
+
+    private final MongoDbConfiguration mongoConf;
+    private final MongoClient mongoClient;
+
+    public MongoDbFruitRepository(MongoDbConfiguration mongoConf,  // <2>
+                                  MongoClient mongoClient) {  // <3>
+        this.mongoConf = mongoConf;
+        this.mongoClient = mongoClient;
+    }
+
+    @Override
+    public void save(@NonNull @NotNull @Valid Fruit fruit){
+        getCollection().insertOne(fruitToDocument(fruit));
+    }
+
+    @Override
+    @NonNull
+    public List<Fruit> list() {
+        List<Fruit> list = new ArrayList<>();
+        try (MongoCursor<Document> cursor = getCollection().find().iterator()) {
+            while (cursor.hasNext()) {
+                documentToFruit(cursor.next()).ifPresent(list::add);
+            }
+        }
+        return list;
+    }
+
+    @NonNull
+    private Optional<Fruit> documentToFruit(@NonNull Document document) {
+        try {
+            String name = document.getString("name");
+            if (name == null) {
+                return Optional.empty();
+            }
+            return Optional.of(new Fruit(name, document.getString("description")));
+        } catch (ClassCastException e) {
+            return Optional.empty();
+        }
+    }
```

Comment:
There is no need for this kind of manual marshalling. You can call `getCollection(name, Fruit.class)` to get a collection that is capable of encoding and ecoding `Fruit`

See https://mongodb.github.io/mongo-java-driver/3.5/bson/pojos/

Reply:
I have changed to use `getCollection(name, Fruit.class)` as you suggest. The drawback of this change is that are forced to have an empty constructor with setters and I cannot have final fields in the `Fruits` POJO.

Reply:
You can but you need to annotate the constructor with `BsonCreator`. See https://mongodb.github.io/mongo-java-driver/3.5/bson/pojos/ and "Supporting POJOs without no args constructors"

Reply:
I have changed the code to use `@BsonCreator` and `@BsonProperty`